example/config/.mocharc.js

```diff
@@ -12,5 +12,7 @@ module.exports = {
   reporter: 'spec',
   slow: 75,
   timeout: 2000,
-  ui: 'bdd'
+  ui: 'bdd',
+  watchFiles: ['lib/**/*.js', 'test/**/*.js'],
```

Comment:
Probably it's better to use the canonical names as shown in the documentation:
`watch-files`, `watch-ignore`

Reply:
Fixed in https://github.com/mochajs/mocha/pull/3980/commits/f18a0b66b5df455fc7b1598797e0e3245df52826